modules/042-kube-dns/openapi/config-values.yaml

```diff
@@ -9,6 +9,15 @@ properties:
     items:
       type: string
       pattern: '^[0-9]{1,}\.[0-9]{1,}\.[0-9]{1,}\.[0-9]{1,}(:[0-9]{1,})?$'
+  transportProtocolMode:
+    type: string
+    description: |
+      The mode defines the transport protocol for communicating with the upstream DNS server:
+      - `PreferUDP` - Uses UDP, even if the original request came over TCP. However, if the DNS response exceeded 512 bytes, it will be truncated and marked with the TC bit by upstream, as specified in [RFC 1035](https://datatracker.ietf.org/doc/html/rfc1035.html#section-4.2.1). Then based on [RFC 5966](https://datatracker.ietf.org/doc/html/rfc5966), the request will be resent over TCP. If there are any other UDP issues, the request will not be resent over TCP.
+      - `ForceTCP` - Always uses TCP, even if the original request comes in over UDP.
+      - `KeepOriginal` - Uses the same protocol (TCP or UDP) as the original request.
```

Comment:
```suggestion
      - `PreferUDP` — UDP is used, even if the original request came over TCP. Note that if the DNS response from the upstream DNS server exceeds 512 bytes and has been truncated and marked with the TC bit ([RFC 1035](https://datatracker.ietf.org/doc/html/rfc1035.html#section-4.2.1)), then it will be resent over TCP ([RFC 5966](https://datatracker.ietf.org/doc/html/rfc5966)). The request will not be resent over TCP if there are other UDP issues.
      - `ForceTCP` — Always uses TCP, even if the original request is over UDP.
      - `KeepOriginal` — Uses the same protocol (TCP or UDP) as the original request.
```